app/uk/gov/hmrc/gform/sharedmodel/formtemplate/AuthConfig.scala

```diff
@@ -97,17 +102,18 @@ object AuthConfig {
     // format: OFF
     val reads = Reads[AuthConfig] { json =>
       for {
-        authModule       <- (json \ "authModule").validate[AuthConfigModule]
-        regimeId         <- (json \ "regimeId").validateOpt[RegimeId]
-        serviceId        <- (json \ "serviceId").validateOpt[ServiceId]
+        authModule <- (json \ "authModule").validate[AuthConfigModule]
+        regimeId <- (json \ "regimeId").validateOpt[RegimeId]
+        serviceId <- (json \ "serviceId").validateOpt[ServiceId]
+        agentAccess <- (json \ "agentAccess").validateOpt[AgentAccess]
         enrolmentSection <- (json \ "enrolmentSection").validateOpt[EnrolmentSection]
-        result           <- (authModule, regimeId, serviceId, enrolmentSection) match {
-          case (AuthConfigModule(`eeittAuth`), Some(_), None, None) => EEITTAuthConfig.format.reads(json)
-          case (AuthConfigModule(`hmrcAuth`), None,    None,    None)    => HMRCAuthConfigWithAuthModule.format.reads(json)
-          case (AuthConfigModule(`hmrcAuth`), None,    Some(_), None)    => HMRCAuthConfigWithServiceId.format.reads(json)
-          case (AuthConfigModule(`hmrcAuth`), Some(_), Some(_), None)    => HMRCAuthConfigWithRegimeId.format.reads(json)
-          case (AuthConfigModule(`hmrcAuth`), None,    Some(_), Some(_)) => HMRCAuthConfigWithEnrolment.format.reads(json)
-          case (AuthConfigModule(`hmrcAuth`), Some(_), Some(_), Some(_)) => HMRCAuthConfig.format.reads(json)
+        result <- (authModule, agentAccess, regimeId, serviceId, enrolmentSection) match {
+          case (AuthConfigModule(eeittAuth), None, Some(_), None, None) => EEITTAuthConfig.format.reads(json)
```

Comment:
I'm not sure removing of backticks`` ` `` has desired effect. 

`AuthConfigModule(eeittAuth)` is as good as `AuthConfigModule(_)`